src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp

```diff
@@ -111,6 +137,95 @@ bool ov::pass::ReverseShapeAndTypeInfer::run_on_model(const std::shared_ptr<ov::
                 }
             }
             is_changed |= inherit_output_type(op, {0, 1});
+        } else if (std::dynamic_pointer_cast<Concat>(op)) {
+            std::vector<size_t> inp_idxs(op->get_input_size());
+            std::iota(inp_idxs.begin(), inp_idxs.end(), 0);
+            is_changed |= inherit_output_rank(op, inp_idxs);
+            is_changed |= inherit_output_type(op, inp_idxs);
+        } else if (std::dynamic_pointer_cast<Slice>(op)) {
+            is_changed |= inherit_output_rank(op, {0});
+            is_changed |= inherit_output_type(op, {0});
+        } else if (std::dynamic_pointer_cast<Squeeze>(op)) {
+            auto in0_rank = op->get_input_partial_shape(0).rank();
+            if (output_shape.rank().is_static() && in0_rank.is_dynamic() && op->get_input_size() > 1) {
+                auto in1_pshape = op->get_input_partial_shape(1);
+                if (in1_pshape.is_static()) {
+                    auto num_dims = in1_pshape.size() == 0 ? 1 : in1_pshape[0].get_length();
+                    op->get_input_tensor(0).m_partial_shape =
+                        PartialShape::dynamic(output_shape.rank().get_length() - num_dims);
```

Comment:
what if we have squeezing to a scalar from 1D tensor with one element, we will get `output_shape.rank().get_length() - num_dims = 0`

Reply:
Fixed